### PR TITLE
Run CanCloseAsync with DispatcherPriority Input.

### PR DESCRIPTION
**Pull request by Stannieman**
This fixes https://github.com/AvaloniaUI/Avalonia/issues/12439

### CODE REVIEW

File: src/Caliburn.Micro.Platform/Platforms/netcore-avalonia/WindowConductor.cs [hunk 1]
```diff
@@ -98,24 +97,26 @@ private async void Closing(object sender, CancelEventArgs e)
...
+            {
+                var canClose = await ((IGuardClose)model).CanCloseAsync(CancellationToken.None);
+
+                if (!canClose)
+                    return;
+
+                actuallyClosing = true;
+                view.Close();
+
+                //if (cachedDialogResult == null)
```
Comment (vb2ae) [inline src/Caliburn.Micro.Platform/Platforms/netcore-avalonia/WindowConductor.cs:110]: could you remove the commented out code. It looks good

**Comment by Stannieman** [comment]
@vb2ae please wait with merging this, I found that it can hang the app when it tries to close a window that was shown as dialog.

**Comment by Mrxx99** [comment]
> @vb2ae please wait with merging this, I found that it can hang the app when it tries to close a window that was shown as dialog.

You can click the convert to draft button for this and when everything is ready convert it back by clicking "ready for review"

**Comment by Stannieman** [comment]
Thanks, I did not know it exists, the button is quite hidden 😅

**Comment by Stannieman** [comment]
Goddammit, that was a rookie mistake. It should be InvokeAsync instead of Invoke cause Invoke is blocking and that is not good.

**Pull request merged**